lib/iknow_cache.rb

```diff
@@ -1,6 +1,42 @@
 # frozen_string_literal: true
 
 class IknowCache
+  Config = Struct.new(:logger, :cache)
+
+  class ConfigWriter
+    def initialize(config)
+      @config = config
+    end
+
+    def logger(logger)
+      @config.logger = logger
+    end
+
+    def cache(cache)
+      @config.cache = cache
+    end
+  end
+
+  def self.configured?
+    @configured
+  end
+
+  def self.configure!(&block)
+    raise ArgumentError.new("Already configured!") if configured?
+
+    conf = Config.new
+    ConfigWriter.new(conf).instance_eval(&block)
+
+    @logger = conf.logger
```

Comment:
Should there be an exposed `IknowCache.config`, that `.logger` and `.cache` delegate to?

Reply:
I considered it but didn't get around to it. You're right that it is nicer though. Added.